entrypoint.sh

```diff
@@ -6,33 +6,45 @@ export HOME="/github/workspace"
 export NVM_DIR="/github/workspace/nvm"
 export WRANGLER_HOME="/github/workspace"
 
-# h/t https://github.com/elgohr/Publish-Docker-Github-Action
-sanitize() {
-  if [ -z "${1}" ]
-  then
-    >&2 echo "Unable to find ${2}. Did you add a GitHub secret called key ${2}, and pass in secrets.${2} in your workflow?"
-    exit 1
-  fi
-}
-
 curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.0/install.sh | bash
+
+# shellcheck source=/dev/null
 [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+# shellcheck source=/dev/null
 [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"
 
 mkdir -p "$HOME/.wrangler"
 chmod -R 770 "$HOME/.wrangler"
 
-sanitize "${INPUT_EMAIL}" "email"
-sanitize "${INPUT_APIKEY}" "apiKey"
+if [ -n "$INPUT_APITOKEN" ]
+then
+  echo "Using API token authentication"
+  export CF_API_TOKEN="$INPUT_APITOKEN"
+fi
+
+if [ -n "$INPUT_APIKEY" ] || [ -n "$INPUT_EMAIL" ]
+then
+  echo "Using legacy email and API key authentication"
+  export CF_EMAIL="$INPUT_EMAIL"
+  export CF_API_KEY="$INPUT_APIKEY"
+fi
 
-export CF_EMAIL="$INPUT_EMAIL"
-export CF_API_KEY="$INPUT_APIKEY"
+if [ -z "$INPUT_APITOKEN" ] && [ -z "$INPUT_EMAIL" ] && [ -z "$INPUT_APIKEY" ]
+then
+  >&2 echo "Unable to find authentication details. Please pass in an 'apiToken' as an input to the action, or a legacy 'apiKey' and 'email'."
+  exit 1
+fi
 
-npm i @cloudflare/wrangler -g
+if [ -z "$INPUT_WRANGLERVERSION" ]
+then
+  npm i @cloudflare/wrangler -g
+else
+  npm i "@cloudflare/wrangler@$INPUT_WRANGLERVERSION" -g
+fi
 
-if ! [ -z "$INPUT_WORKINGDIRECTORY" ]
+if [ -n "$INPUT_WORKINGDIRECTORY" ]
```

Comment:
what is -n vs -z?

Reply:
may be worth adding some comments.

Reply:
Added comments for each of these conditionals, thanks!